File: ee-schkola/src-gen/main/kotlin/ee/schkola/auth/AuthApiBase.kt
package ee.schkola.auth

import ee.schkola.SchkolaBase
import ee.schkola.person.Profile
import java.util.*


open class Account : SchkolaBase {
    val username: String
    val password: String
    val email: String
    val disabled: Boolean
    val lastLoginAt: Date
    val profile: Profile


    constructor(id: String = "", username: String = "", password: String = "", email: String = "", disabled: Boolean = false, 
                lastLoginAt: Date = Date(), profile: Profile = Profile()) : super(id) {
        this.username = username
        this.password = password
        this.email = email
        this.disabled = disabled
        this.lastLoginAt = lastLoginAt
        this.profile = profile
    }

    companion object {
        val EMPTY = Account()
    }
}

File: ee-schkola/src-gen/main/go/ee/schkola/auth/AuthApiBase.go
package auth

import (
    "ee/schkola"
    "ee/schkola/person"
    "time"
)

type Account struct {
    username string
    password string
    email string
    disabled bool
    lastLoginAt *time.Time
    profile *person.Profile
    *schkola.SchkolaBase
}

func NewAccount(username string, password string, email string, disabled bool, lastLoginAt *time.Time, profile *person.Profile, 
                SchkolaBase *schkola.SchkolaBase) (ret *Account, err error) {
    ret = &Account{
        username: username,
        password: password,
        email: email,
        disabled: disabled,
        lastLoginAt: lastLoginAt,
        profile: profile,
        SchkolaBase: SchkolaBase,
    }
    
    return
}









